pkg/ebpf/controlplane/controller.go

```diff
@@ -112,69 +186,22 @@ func (ctrl *Controller) Stop() error {
 	return nil
 }
 
+// RegisterSignal registers a signal handler for the given signal ID
+func (ctrl *Controller) RegisterSignal(signalID events.ID, handler SignalHandler) {
+	ctrl.signalHandlers[signalID] = handler
```

Comment:
I guess it should take care of existing handlers, erring when an override would occur.

Reply:
Make total sense... good catch... tks